.NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/TimexResolver.cs

```diff
@@ -458,36 +560,81 @@ private static List<Resolution.Entry> ResolveDateTime(TimexProperty timex, DateO
+            if (year == Constants.InvalidValue)
+            {
+                result.Add(MonthWeekDateRange(referYear - 1, month, weekOfMonth));
+                result.Add(MonthWeekDateRange(referYear, month, weekOfMonth));
+            }
+            else
+            {
+                result.Add(MonthWeekDateRange(year, month, weekOfMonth));
+            }
+
+            return result;
+        }
+
         private static List<Resolution.Entry> ResolveDateTimeRange(TimexProperty timex, DateObject date)
         {
             if (timex.PartOfDay != null)
             {
-                var dateValue = TimexValue.DateValue(timex);
+                var dateValues = GetDateValues(timex, date);
                 var timeRange = PartOfDayTimeRange(timex);
-                return new List<Resolution.Entry>
+                var result = new List<Resolution.Entry> { };
+                foreach (string dateValue in dateValues)
                 {
-                    new Resolution.Entry
-                    {
-                        Timex = timex.TimexValue,
-                        Type = "datetimerange",
-                        Start = $"{dateValue} {timeRange.Item1}",
-                        End = $"{dateValue} {timeRange.Item2}",
-                    },
-                };
+                    result.Add(
+                        new Resolution.Entry
+                        {
+                            Timex = timex.TimexValue,
+                            Type = "datetimerange",
+                            Start = $"{dateValue} {timeRange.Item1}",
+                            End = $"{dateValue} {timeRange.Item2}",
+                        });
+                }
+
+                return result;
             }
             else
             {
                 var range = TimexHelpers.ExpandDateTimeRange(timex);
-                return new List<Resolution.Entry>
+                var startDateValues = GetDateValues(range.Start, date);
+                var endDateValues = GetDateValues(range.End, date);
+                var result = new List<Resolution.Entry> { };
+                foreach (var dateRange in startDateValues.Zip(endDateValues, (n, w) => new { start = n, end = w }))
                 {
-                    new Resolution.Entry
-                    {
-                        Timex = timex.TimexValue,
-                        Type = "datetimerange",
-                        Start = $"{TimexValue.DateValue(range.Start)} {TimexValue.TimeValue(range.Start, date)}",
-                        End = $"{TimexValue.DateValue(range.End)} {TimexValue.TimeValue(range.End, date)}",
-                    },
-                };
+                    result.Add(
+                        new Resolution.Entry
+                        {
+                            Timex = timex.TimexValue,
+                            Type = "datetimerange",
+                            Start = $"{dateRange.start} {TimexValue.TimeValue(range.Start, date)}",
```

Comment:
Can you change these data string concatenations/internpolations to be a methid? This was it's easier to change the format in a single place in the future.

If no such method exists, you could call it FormatResolvedDateValue().